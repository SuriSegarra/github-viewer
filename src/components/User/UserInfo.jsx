import React from 'react';
import PropTypes from 'prop-types';

const UserInfo = ({ login, followers, following, url, avatar }) => (
  <figure>
    <img src={avatar} alt={login}/>
    <figcaption>
      <h1>{login}</h1>
      <p>Followers:{followers}</p>
      <p>Following:{following}</p>
      <p>Profile:{url}</p>
    </figcaption>
  </figure>
);

UserInfo.propTypes = {
  login: PropTypes.string.isRequired,
  followers: PropTypes.string.isRequired,
  following: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  avatar: PropTypes.string.isRequired

};

export default UserInfo;
