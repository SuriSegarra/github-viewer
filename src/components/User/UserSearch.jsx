import React from 'react';
import PropTypes from 'prop-types';

const userSearch = ({ username, onUserChange, onUserSubmit }) => (
  <>
    <input type='text' name='username' value={username} onChange={onUserChange}/>
    <input type='button' name='search' value='Search' onClick={onUserSubmit}></input>
  </>
);

userSearch.propTypes = {
  username: PropTypes.string.isRequired,
  onUserChange: PropTypes.func.isRequired,
  onUserSubmit: PropTypes.func.isRequired
};

export default userSearch;
