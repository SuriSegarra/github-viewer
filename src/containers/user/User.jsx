import React, { Component } from 'react';
import UserInfo from '../../components/User/UserInfo';
import { fetchUser } from '../../services/github';

export default class User extends Component {
  state = {
    user: '',
    userState: {}
  }

  componentDidMount() {
    fetchUser()
    .then(user => this.setState({
      userState: user,
      
    }))
  }
  render() {
    return (
      <div>
        <UserInfo login={this.userState.login} followers={this.userState.followers} following={this.userState.} />
      </div>
    );
  }
}
