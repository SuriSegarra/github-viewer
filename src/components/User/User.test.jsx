import React from 'react';
import { shallow } from 'enzyme';
import User from './User';

describe('User', () => {
  it('matches a user snapshot', () => {
    const wrapper = shallow(<User login='Surelis Segarra' 
      followers_url='1' 
      following_url='3' 
      url='https://github.com/SuriSegarra' 
      avatar_url='https://avatars2.githubusercontent.com/u/57576258?v=4' />);
    expect(wrapper).toMatchSnapshot();
  });
});
