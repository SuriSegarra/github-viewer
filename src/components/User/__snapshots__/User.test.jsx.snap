// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User matches a snapshot 1`] = `
<figure>
  <img
    alt="Surelis Segarra"
    src="https://avatars2.githubusercontent.com/u/57576258?v=4"
  />
  <figcaption>
    <p>
      Surelis Segarra
    </p>
    <p>
      Followes:
      1
    </p>
    <p>
      Following:
      3
    </p>
    <p>
      https://github.com/SuriSegarra
    </p>
  </figcaption>
</figure>
`;
